### ISSUE TITLE
Gradle 7.6.1: Unable to load class `kotlin.enums.EnumEntriesKt` (Kotlin DSL)

### ISSUE THREAD

**Issue by felixscheinost** [craigatk/opentelemetry-gradle-plugin#92]
Using Gradle 7.6.1 with Kotlin buildscript yields the following error:

```
An exception occurred applying plugin request [id: 'com.atkinsondev.opentelemetry-build', version: '1.6.0']
> Failed to apply plugin 'com.atkinsondev.opentelemetry-build'.
   > Could not create an instance of type com.atkinsondev.opentelemetry.build.OpenTelemetryBuildPluginExtension.
      > kotlin/enums/EnumEntriesKt
```

Version 1.5.0 of the plugin works fine.

I tested in a dummy project that changing from `build.gradle.kts` to `build.gradle` also fixes the problem.

Seems like this plugin has a transitive dependency on a newer Kotlin version (EnumEntries is a newer addition to the Kotlin stdlib?) than the one that Gradle 7.6.1 ships with.

This table lists a list of Gradle version and embedded Kotlin version: https://docs.gradle.org/current/userguide/compatibility.html

Depending on Kotlin 1.9.10 makes this plugin compatible only with Gradle >= 8.3.

I tried upgrading the embedded Kotlin version in the `buildscript` block but that also gave an error as Gradle has a strict dependency constraint for that dependency.

**Comment by felixscheinost** [comment]
Ah I see renovate bot did the update. I think for a Gradle plugin the Kotlin version can't be updated automatically like that.

**Comment by craigatk** [comment]
Thanks for the heads up about this Kotlin version issue! I downgraded the project to Kotlin 1.7.10 to match the version of Kotlin bundled with the version of Gradle this project uses (7.6) and released plugin version 1.6.1 with the changes.

**Comment by felixscheinost** [comment]
Cool, thanks!
